### PR TITLE
Use a circle for the ball in Breakout

### LINKED ISSUE #4324: Use a circle for the ball in Breakout
Once #3730 is merged, we should represent the ball as a circle, rather than a square. We can probably stick with the AABB collision though for simplicity.

### PR DESCRIPTION
**Pull request by shaderduck**
# Objective

Fixes #4324 

## Solution

Used circle from shapes example in #3730 

---

### Changed
Circle shape instead of square shape sprite for ball.
AABB collision unchanged.

### CODE REVIEW

File: examples/games/breakout.rs [hunk 1]
```diff
@@ -202,16 +210,14 @@ fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
     commands
         .spawn()
         .insert(Ball)
-        .insert_bundle(SpriteBundle {
+        .insert_bundle(MaterialMesh2dBundle {
+            mesh: meshes.add(shape::Circle::default().into()).into(),
```
Comment (alice-i-cecile) [inline examples/games/breakout.rs:214]:
```suggestion
            /// Meshes define the shape of your objects
            mesh: meshes.add(shape::Circle::default().into()).into(),
```

File: examples/games/breakout.rs [hunk 2]
```diff
@@ -202,16 +210,14 @@ fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
...
+            material: materials.add(ColorMaterial::from(BALL_COLOR)),
```
Comment (alice-i-cecile) [inline examples/games/breakout.rs:215]:
```suggestion
            // Materials define the texture of your objects
            // In this case, we're just using a flat color
            material: materials.add(ColorMaterial::from(BALL_COLOR)),
```

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Nice work :) Some small suggestions for comments to help this example teach beginners the basic terminology.

**Comment by alice-i-cecile** [comment]
bors try

**Comment by Nilirad** [comment]
Is the z value in `BALL_STARTING_POSITION` (or at least the comment above the constant) still relevant now that we have a potential 2D Mesh/Sprite overlap, instead of a sprite/sprite overlap?

**Comment by alice-i-cecile** [comment]
Unsure 🤔 I haven't played with the mesh / sprite overlap rules.

**Review by Nilirad — COMMENTED** [review]
I found out that the order in which sprites and 2D meshes are drawn is undefined, just like sprites, so the z layering should remain.

<details>
<summary>Example code</summary>

```rust
use bevy::{math::const_vec3, prelude::*, sprite::MaterialMesh2dBundle};

const PADDLE_SIZE: Vec3 = const_vec3!([120.0, 20.0, 0.0]);

const BALL_STARTING_POSITION: Vec3 = const_vec3!([0.0, -50.0, 1.0]);
const BALL_SIZE: Vec3 = const_vec3!([30.0, 30.0, 0.0]);

const PADDLE_COLOR: Color = Color::rgb(0.3, 0.3, 0.7);
const BALL_COLOR: Color = Color::rgb(1.0, 0.5, 0.5);

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_startup_system(setup)
        .run();
}

fn setup(
    mut commands: Commands,
    mut meshes: ResMut<Assets<Mesh>>,
    mut materials: ResMut<Assets<ColorMaterial>>,
) {
    // Camera
    commands.spawn_bundle(OrthographicCameraBundle::new_2d());

    // Paddle
    commands.spawn().insert_bundle(SpriteBundle {
        transform: Transform {
            translation: BALL_STARTING_POSITION,
            scale: PADDLE_SIZE,
            ..default()
        },
        sprite: Sprite {
            color: PADDLE_COLOR,
            ..default()
        },
        ..default()
    });

    // Ball
    commands.spawn().insert_bundle(MaterialMesh2dBundle {
        mesh: meshes.add(shape::Circle::default().into()).into(),
        material: materials.add(ColorMaterial::from(BALL_COLOR)),
        transform: Transform {
            scale: BALL_SIZE,
            translation: BALL_STARTING_POSITION,
            ..default()
        },
        ..default()
    });
}

```

</details>

So the only thing to do is to change the comment above `BALL_STARTING_POSITION` to something like:

```rust
// We set the z-value of the ball to 1 so it renders on top in the case of overlap.
const BALL_STARTING_POSITION: Vec3 = const_vec3!([0.0, -50.0, 1.0]);
```

**Comment by alice-i-cecile** [comment]
@shaderduck can you rebase this? This would be nice to merge in.

**Comment by alice-i-cecile** [comment]
Closing in favor of #5657 :) Thanks for getting this started!

**Pull request closed without merging**